### PR TITLE
[gh-3][Infrastructure] As a user, I can access the Terraform cloud

### LINKED ISSUE #3: [Infrastructure] As a user, I can access the Terraform cloud
## Why

Integrate the Terraform cloud to help visualize the checks, deployments, and set variables.

## Acceptance Criteria

- Terraform Cloud is set and goes live.
- Deployments and Plans can be managed by Terraform cloud
- 3 workspaces should be created:
  - `workspace-production`: triggers run from `main` branch and `core` folder
  - `workspace-staging`: triggers run from. `develop` branch and `core` folder
  - `workspace-shared`: trigger run from `develop` branch and `shared` folder

### PR DESCRIPTION
**Pull request by nvminhtue**
- Close #3 

## What happened 👀

Integrate into the Terraform cloud
- The workspace can be found [here](https://app.terraform.io/app/nvminhtue/workspaces)
- There are 3 different workspaces
  - [production](https://app.terraform.io/app/nvminhtue/workspaces/devops-ic-production)
  - [staging](https://app.terraform.io/app/nvminhtue/workspaces/devops-ic-staging)
  - [shared](https://app.terraform.io/app/nvminhtue/workspaces/devops-ic-shared)

## Insight 📝

- Add the `.tool-versions` file matched with the current and latest terraform version
- Provide a more detailed README

## Proof Of Work 📹
Logged-in
image

Integrated 3 workspaces into Terraform cloud
image
image

Resources has successfully planned on local
image

### CODE REVIEW

File: core/providers.tf [hunk 1]
```diff
@@ -1,4 +1,11 @@
 terraform {
+  cloud {
+    organization = "nvminhtue"
+
+    workspaces {
+      tags = ["aws-infrastructure"]
+    }
+  }
```
Comment (liamstevens111) [inline core/providers.tf:8]: What about moving this part out of `providers.tf` into a `main.tf` (same for shared)

```suggestion
  cloud {
    organization = "nvminhtue"

    workspaces {
      tags = ["aws-infrastructure"]
    }
  }
  
  # Terraform version
  required_version = "~> 1.6"
```

We can also add the required `terraform` version in the outer terraform block too (you need to create this block again in `main`). It seems you are using the latest (`1.6`) so I can't comment on any breaking changes or differences in the new versions though.

**Review by liamstevens111 — APPROVED** [review]
One comment, otherwise looks good 👍 

FYI If the Terraform Cloud project/workspaces are on your personal account I don't think we could access them.

pre-approved for now.

File: core/providers.tf (hunk 1 above)
Comment (nvminhtue) [reply core/providers.tf:8]: I don't see any benefit to the main file though, added the `required_version` in [5ee9924](https://github.com/nvminhtue/tyson_devops_ic/pull/20/commits/5ee9924687402849599d746fbe9911763778d325)

Comment (liamstevens111) [reply core/providers.tf:8]: I actually don't know the real reasoning, seems we do it on projects though, ie

https://github.com/nimblehq/easyhotel-infra-aws/blob/develop/core/main.tf

@bterone Ill let you decide on this one, I am also fine with it where it is.

Comment (bterone) [reply core/providers.tf:8]: I've seen some projects even put versions in a separate [versions.tf](https://github.com/terraform-aws-modules/terraform-aws-vpc/blob/bf9a89bf447a9c866dc0d30486aec5a24dbe2631/versions.tf) file 

or even some having a dedicated [terraform.tf ](https://github.com/trussworks/terraform-layout-example/blob/main/orgname-id/admin-global/terraform.tf)file for this 

Under the [AWS best practices](https://github.com/aws-samples/aws-terraform-best-practices#standard-module-structure) for terraform, they keep the terraform block and provider blocks under the `providers.tf`, so I'm also fine with this approach 👍

**Review by bterone — APPROVED** [review]
LGTM

**Pull request merged**